server/src/main/java/org/opensearch/search/fetch/subphase/MatchedQueriesPhase.java

```diff
@@ -65,6 +67,17 @@ public FetchSubPhaseProcessor getProcessor(FetchContext context) throws IOExcept
         if (context.parsedPostFilter() != null) {
             namedQueries.putAll(context.parsedPostFilter().namedFilters());
         }
+        if (context.rescore() != null) {
+            for (RescoreContext rescoreContext : context.rescore()) {
+                if (rescoreContext instanceof QueryRescorer.QueryRescoreContext) {
```

Comment:
Lets please update `RescoreContext` interface here to provide parsedQuery(), then we don't need the type checks.